tests/servers/tests.py

```diff
@@ -57,29 +56,57 @@ def test_protocol(self):
+        for all simple views and streaming responses where an iterable with
+        length of one passed.
+
+        If we cannot detect a content length we explicitly set the `Connection`
+        header to `close` to notify the client that we do not actually support
+        it.
         """
         conn = HTTPConnection(LiveServerViews.server_thread.host, LiveServerViews.server_thread.port, timeout=1)
         try:
-            conn.request('GET', '/example_view/', headers={'Connection': 'keep-alive'})
-            response = conn.getresponse().read()
-            conn.request('GET', '/example_view/', headers={'Connection': 'close'})
-            # macOS may give ConnectionResetError.
-            with self.assertRaises((RemoteDisconnected, ConnectionResetError)):
-                try:
-                    conn.getresponse()
-                except ConnectionAbortedError:
-                    if sys.platform == 'win32':
-                        self.skipTest('Ignore nondeterministic failure on Windows.')
+            conn.request('GET', '/streaming_example_view/', headers={'Connection': 'keep-alive'})
+            response = conn.getresponse()
+            self.assertTrue(response.will_close)
+            response.read()
+            self.assertEqual(response.status, 200)
+            self.assertEqual(response.getheader('Connection'), 'close')
+            conn.request('GET', '/streaming_example_view/', headers={'Connection': 'close'})
+            response = conn.getresponse()
+            self.assertTrue(response.will_close)
+            response.read()
+            self.assertEqual(response.status, 200)
+            self.assertEqual(response.getheader('Connection'), 'close')
+        finally:
+            conn.close()
+
+    def test_keep_alive_on_connection_with_content_length(self):
+        """
+        See `test_closes_connection_without_content_length` for details. This
+        is a follow up test, which ensure that we do not close the connection
+        if not needed, hence allowing us to take advantage of keep-alive.
+        """
+        conn = HTTPConnection(LiveServerViews.server_thread.host, LiveServerViews.server_thread.port)
+        try:
+            conn.request('GET', '/example_view/', headers={"Connection": "keep-alive"})
+            response = conn.getresponse()
+            self.assertFalse(response.will_close)
+            response.read()
+            self.assertEqual(response.status, 200)
+            self.assertIsNone(response.getheader('Connection'))
+            conn.request('GET', '/example_view/', headers={"Connection": "close"})
+            response = conn.getresponse()
+            self.assertFalse(response.will_close)
+            response.read()
+            self.assertEqual(response.status, 200)
+            self.assertIsNone(response.getheader('Connection'))
         finally:
             conn.close()
-        self.assertEqual(response, b'example view')
```

Comment:
You didn't want to capture the response content and assert that here? (Just wondering your thoughts...)